app/src/main/java/com/github/se/orator/ui/profile/ProfileScreen.kt

```diff
@@ -232,38 +271,215 @@ fun ProfileScreen(navigationActions: NavigationActions, profileViewModel: UserPr
+                              Modifier.fillMaxWidth()
+                                  .fillMaxHeight()
+                                  .background(colors.onPrimary)
+                                  .padding(AppDimensions.paddingMedium)) {
+                            // Example content for the stats screen
+                            // TODO : Replace this with graph screens worked on in #168
+                            Text("Graph or Stats content goes here")
+                          }
+                    }
               }
-                  ?: run {
+              Spacer(modifier = Modifier.height(AppDimensions.paddingSmall))
+
+              Column(
+                  modifier =
+                      Modifier.fillMaxSize()
+                          .padding(innerPadding)
+                          .padding(AppDimensions.paddingMedium)
+                          .testTag(("offline_recordings_column"))) {
+
+                    // Title for Offline Recordings
                     Text(
-                        text = "Loading profile...",
-                        style = AppTypography.bodyLargeStyle,
-                        modifier = Modifier.testTag("loading_profile_text"))
+                        text = "My Offline Recordings",
+                        style =
+                            TextStyle(
+                                fontSize = 18.sp,
+                                fontFamily = FontFamily(Font(R.font.poppins_black)),
+                                color = colors.onSurface),
+                        maxLines = 1,
+                        overflow = TextOverflow.Ellipsis,
+                        modifier =
+                            Modifier.padding(vertical = AppDimensions.paddingSmall)
+                                .testTag("offline_recordings_title"))
+
+                    // Display saved recordings
+                    if (savedRecordings.isNotEmpty()) {
+                      savedRecordings.forEach { audioFile ->
+                        AudioRecordingPlaceholder(
+                            fileName = audioFile.name,
+                            onPlayClicked = { audioRecorder.playAudio(audioFile) })
+                      }
+                    } else {
+                      Text(
+                          text = "No offline recordings available",
+                          style = MaterialTheme.typography.bodySmall,
+                          color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.6f),
+                          modifier =
+                              Modifier.padding(AppDimensions.paddingSmall)
+                                  .testTag("no_recordings_text"))
+                    }
                   }
             }
+            ?: run {
+              Text(
+                  text = "Loading profile...",
+                  style = AppTypography.bodyLargeStyle,
+                  modifier = Modifier.testTag("loading_profile_text"))
+            }
```

Comment:
I fact, as you already noticed, this code is never executed and can hence be removed.